pkg/workload/schemachange/operation_generator.go

```diff
@@ -3689,7 +3689,11 @@ func isClusterVersionLessThan(
 	ctx context.Context, tx pgx.Tx, targetVersion roachpb.Version,
 ) (bool, error) {
 	var clusterVersionStr string
-	row := tx.QueryRow(ctx, `SHOW CLUSTER SETTING version`)
+	// Directly querying the cluster version with SHOW CLUSTER
+	// SETTING version isn't safe, since it will wait for any
```

Comment:
Could you point me at that code? Is that new behavior? I'm also seeing a flake in https://github.com/cockroachdb/cockroach/issues/99459 that seems like the same thing, but we had that test for years so I'm wondering which PR changed semantics and whether that was intentional.

Reply:
Not PR author, just looking at the code, but i see that show cluster setting will fail if local value != queried value from settings table (by searching the test error), while show settings seem to be delegating to querying crdb_internal.cluster_settings which doesn't check local values? That's my best guess.

Reply:
This bit basically
https://github.com/cockroachdb/cockroach/blob/a96106157006a8c93c0d7c22d4246988f4f8536d/pkg/sql/show_cluster_setting.go#L59-L103
which is not applicable for show all settings.

Reply:
What I don't understand is that in both #99115 and #99459, I don't see a migration taking more than a few seconds (let alone 2mins) to finish around the time of the test failure. Am I missing something?

Reply:
IMHO once version querying code is fixed, we'll see the real issue because version that test expects will diverge and give us more sensible failure. Same in other test.